quickwit/quickwit-serve/src/lib.rs

```diff
@@ -1145,6 +1147,7 @@ fn setup_indexer_pool(
                     } else {
                         let client = IndexingServiceClient::tower()
                             .stack_layer(INDEXING_GRPC_CLIENT_METRICS_LAYER.clone())
+                            .stack_layer(TimeoutLayer::new(Duration::from_secs(30)))
```

Comment:
if you use the layer in tower, I think here is a helper function that goes 
`.timeout(duration)`

Reply:
Yes, with the manual TimeoutLayer we don't have access to that, but internally it does the same

```rust
self.layer(crate::timeout::TimeoutLayer::new(timeout))
```